codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/service/ServiceStubGenerator.kt

```diff
@@ -0,0 +1,82 @@
+package software.amazon.smithy.kotlin.codegen.service
+
+import software.amazon.smithy.kotlin.codegen.core.KotlinDelegator
+import software.amazon.smithy.kotlin.codegen.core.KotlinDependency
+import software.amazon.smithy.kotlin.codegen.core.RuntimeTypes
+
+class ServiceStubGenerator(
+    private val packageName: String,
+    private val delegator: KotlinDelegator,
+) {
+
+    fun render() {
+        renderMainFile()
+        renderProtocolModule()
+        renderAuthModule()
+        renderConstraintValidators()
+        renderRouting()
+        renderPerOperationHandlers()
+    }
+
+    // Writes `Main.kt` that boots the embedded Ktor service.
+    private fun renderMainFile() {
+        delegator.useFileWriter("Main.kt", packageName) { writer ->
+
+            writer.addImport(RuntimeTypes.KtorServerCore.embeddedServer)
+            writer.addImport(RuntimeTypes.KtorServerNetty.Netty)
+            writer.dependencies.addAll(KotlinDependency.KTOR_LOGGING_BACKEND.dependencies)
```

Comment:
I think this is fine for now, but you might be missing an abstraction that will allow `ServiceStubGenerator` to generate non-Ktor services in the future. We might need some `ServiceStubSettings` that the user can configure to choose between Ktor and other options, as well as configure Ktor generation